src/main/java/com/cognifide/aemrules/java/checks/resourceresolver/close/ResourceResolverTryWithResourcesCheck.java

```diff
@@ -88,8 +89,8 @@ public void visitAssignmentExpression(AssignmentExpressionTree tree) {
   @Override
   public void visitTryStatement(TryStatementTree tree) {
     insideTryStatement = true;
-    tree.resources().stream()
-        .map(resource -> resource.simpleName().name())
+    tree.resourceList().stream()
+        .map(resource -> ((VariableTreeImpl) resource).simpleName().name())
```

Comment:
What's the reason behind referring to an implementation type here and casting?

Reply:
This was the only way I found to get that prop.

Reply:
`tree.resourceList()` returns a `ListTree<Tree>` and the method you're trying to use here doesn't exist at such a high interface level. Still, you don't need to use the specific implementation type here. As far as I can tell, the interface `VariableTree` should be sufficient

```
         // We're only interested in variables so filter out VariableTree instances from a list of Tree. Not all trees have names.
        .filter(resource -> resource instanceof VariableTree)
        .map(VariableTree.class::cast)
        .map(resource -> resource.simpleName().name())
        .forEach(resourceResolversInTryWithResources::add);
```

I'm trying to think of a unit test case that would show us where this cast could fail without the filter.

Reply:
I can't think of a way to pass a syntax tree that doesn't cast to `VariableTree` inside a `try` statement's resource list that is parse-able as valid Java, as per https://docs.oracle.com/javase/specs/jls/se7/html/jls-14.html#jls-14.20.3

 I wonder why the `resourceTree` method, which appears specific to the `TryStatementTree` doesn't return something more specific than `Tree`. Could be just a matter of consistency.

How about:
```
tree.resourceList().stream()
  // We're iterating over a resource specification in a try-with-resource block
  // so we expect variable trees
  .filter(resource -> resource instanceof VariableTree)
  .map(VariableTree.class::cast)
  .map(resource -> resource.simpleName().name())
  .forEach(resourceResolversInTryWithResources::add);
```

Reply:
This kind of works but it turns out there's a new way to write a try-with-resources block in Java 9 and above.
https://stackoverflow.com/questions/61124346/when-does-sonars-trystatementtreeresourcelist-return-a-tree-that-isnt-a-varia/61124506#61124506

Issue https://github.com/Cognifide/AEM-Rules-for-SonarQube/issues/188 raised to address this.